internal/pkg/policy/monitor.go

```diff
@@ -243,6 +281,7 @@ func (m *monitorT) dispatchPending(ctx context.Context) {
 		// If too many (checkin) responses are written concurrently memory usage may explode due to allocating gzip writers.
 		err := m.limit.Wait(ctx)
 		if err != nil {
+			m.pendingQ.pushFront(s) // context cancelled before sub is handled, put it back
```

Comment:
Would be best to check for the context cancelled error, and update the log to not log `Policy limit error` when its a context cancel.